From 1cb04a74ed9b2f4a05ac21ce1e9a5ca884c9b97e Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Thu, 31 Oct 2013 16:29:29 +0100
Subject: [PATCH 10/29] savevm: qmp_xen_save_devices_state(): use error_setg_file_open()

RH-Author: Laszlo Ersek <lersek@redhat.com>
Message-id: <1383236971-6067-7-git-send-email-lersek@redhat.com>
Patchwork-id: 55196
O-Subject: [RHEL-7 qemu-kvm PATCH 6/8] savevm: qmp_xen_save_devices_state(): use error_setg_file_open()
Bugzilla: 907743
RH-Acked-by: Luiz Capitulino <lcapitulino@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: Luiz Capitulino <lcapitulino@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Acked-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit 1befce9652a1b7cfca0191b3031fae3cbce26ef0)
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
---
 savevm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 savevm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/savevm.c b/savevm.c
index d3bd0ec..1ed7f05 100644
--- a/savevm.c
+++ b/savevm.c
@@ -2419,7 +2419,7 @@ void qmp_xen_save_devices_state(const char *filename, Error **errp)
 
     f = qemu_fopen(filename, "wb");
     if (!f) {
-        error_set(errp, QERR_OPEN_FILE_FAILED, filename);
+        error_setg_file_open(errp, errno, filename);
         goto the_end;
     }
     ret = qemu_save_device_state(f);
-- 
1.7.1

